benchadapt/python/benchadapt/adapters/folly.py

```diff
@@ -12,19 +12,39 @@ class FollyAdapter(BenchmarkAdapter):
 
     result_dir: Path
 
-    def __init__(self, command: List[str], result_dir: Path) -> None:
+    def __init__(
+        self,
+        command: List[str],
+        result_dir: Path,
+        result_fields_override: Dict[str, Any] = None,
+        result_fields_append: Dict[str, Any] = None,
+    ) -> None:
         """
         Parameters
         ----------
         command : List[str]
             A list of strings defining a shell command to run folly benchmarks
         result_dir : Path
             Path to directory where folly results will be populated
+        result_fields_override : Dict[str, Any]
+            A dict of values to override on each instance of `BenchmarkResult`. Useful
+            for specifying metadata only available at runtime, e.g. build info. Applied
+            before ``results_field_append``.
+        results_fields_append : Dict[str, Any]
+            A dict of default values to be appended to `BenchmarkResult` values after
+            instantiation. Useful for appending extra tags or other metadata in addition
+            to that gathered elsewhere. Only applicable for dict attributes. For each
+            element, will override any keys that already exist, i.e. it does not append
+            recursively.
```

Comment:
Is there any way we could depend on these docs being in one place? Or are they subtly different from one another?

Reply:
They're identical. I'm always torn when writing Python docs; there are ways in Sphinx to inherit sections, I think? but not a direct equivalent of roxygen's `@inheritParams` that I've seen. Nor are we actually rendering any of these docs anywhere, so I'm presuming they'll mostly get used by people reading docstrings in the source or their editor, in which case I really want these to actually be documented on the classes users touch, not just the parent class for devs. I'd love a general solution beyond copying and pasting if anyone's got one, but I haven't.